### ISSUE TITLE
Unable to run docker image

### ISSUE THREAD

**Issue by damo1884** [3b1b/manim#253]
OS: Mac 

Docker installed with some errors but completed with

Successfully built 7170abd7c78d
Successfully tagged manim:latest

```
gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DVERSION=1.3.5.dev0 -DHAVE_FREETYPE2 -Iagg2/include -Iagg2/font_freetype -I/usr/include -I/usr/include/freetype2 -I/usr/local/include/python2.7 -c aggdraw.cxx -o build/temp.linux-x86_64-2.7/aggdraw.o
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
In file included from agg2/font_freetype/agg_font_freetype.h:28:0,
                 from aggdraw.cxx:84:
agg2/include/agg_scanline_storage_bin.h: In member function ‘unsigned int agg::scanline_storage_bin::byte_size() const’:
agg2/include/agg_scanline_storage_bin.h:247:38: warning: unused variable ‘sp’ [-Wunused-variable]
                     const span_data& sp = m_spans[span_idx++];
                                      ^
aggdraw.cxx: In function ‘PyObject* draw_new(PyObject*, PyObject*)’:
aggdraw.cxx:691:70: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]
         PyObject* buffer = PyObject_CallMethod(image, "tobytes", NULL);
                                                                      ^
aggdraw.cxx: In function ‘agg::rgba8 getcolor(PyObject*, int)’:
aggdraw.cxx:924:72: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]
         result = PyObject_CallFunction(aggdraw_getcolor_obj, "O", color);
                                                                        ^
aggdraw.cxx: In function ‘PyObject* draw_flush(DrawObject*, PyObject*)’:
aggdraw.cxx:1405:71: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]
     result = PyObject_CallMethod(self->image, "frombytes", "N", buffer);
                                                                       ^
aggdraw.cxx:1405:71: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]
In file included from agg2/font_freetype/agg_font_freetype.h:35:0,
                 from aggdraw.cxx:84:
agg2/include/agg_font_cache_manager.h: In instantiation of ‘void agg::font_cache_manager<FontEngine>::init_embedded_adaptors(const agg::glyph_cache*, double, double) [with FontEngine = agg::font_engine_freetype_int32]’:
aggdraw.cxx:490:60:   required from ‘void draw_adaptor<PixFmt>::drawtext(float*, PyObject*, FontObject*) [with PixFmt = agg::pixel_formats_rgba32<agg::order_rgba32>; PyObject = _object]’
aggdraw.cxx:2253:1:   required from here
agg2/include/agg_font_cache_manager.h:306:17: warning: enumeration value ‘glyph_data_invalid’ not handled in switch [-Wswitch]
                 switch(gl->data_type)
                 ^
In file included from aggdraw.cxx:86:0:
agg2/include/agg_path_storage.h: In function ‘PyObject* draw_ellipse(DrawObject*, PyObject*)’:
agg2/include/agg_path_storage.h:336:25: warning: ‘y’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         *coord_ptr   = y;
                         ^
agg2/include/agg_path_storage.h:179:23: note: ‘y’ was declared here
             double x, y;
                       ^
agg2/include/agg_path_storage.h:335:25: warning: ‘x’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         *coord_ptr++ = x;
                         ^
agg2/include/agg_path_storage.h:179:20: note: ‘x’ was declared here
             double x, y;
                    ^
```

Then trying to start the docker image

```
docker run --rm -v "$PWD/files":/app/files manim example_scenes.py WarpSquare

Traceback (most recent call last):
  File "extract_scene.py", line 14, in <module>
    from constants import *
  File "/app/constants.py", line 32, in <module>
    """)
Exception: 
        Redefine MEDIA_DIR by changing the MEDIA_DIR
        environment constant or by changing
        media_dir.txt to point to a valid directory
        where movies and images will be written
```

I get this error even tho I have updated my MEDIA_DIR in constants.py

Any idea how to get this working in docker? (I also tried directly on mac with no luck)

**Comment by gonzaloarcr** [comment]
Did you rebuild? Did the path exists? Remeber that the path of the container is not the same ass the path on your machine. You can try with /app, for example.

**Comment by cnrmck** [comment]
@damo1884 I had this issue too. The way I fixed it is by making a directory for the media_files and by passing in the correct environment variable that points at it. 

I then couldn't access the output of the scene, so I had to share the media_files directory by way of a volume. Because I'm making frequent changes to the code, I've just decided to share my whole manim directory with the container. If you run the code below, it should do that for you. 

`mkdir media_files`

`docker run --rm --volume="$PWD:/app/" -e MEDIA_DIR=/app/media_files manim example_scenes.py WarpSquare`
